parity/secretstore/server.rs

```diff
@@ -210,6 +216,11 @@ mod server {
 
 			cconf.cluster_config.nodes.insert(self_secret.public().clone(), cconf.cluster_config.listener_address.clone());
 
+			if std::fs::read_dir(&conf.data_path).map_or(false, |mut list| list.next().is_none ()) {
```

Comment:
This would be more readable if you could do instead: 

```rust
let db_path = Path::new(&conf.data_path).join(VERSION_FILE_NAME);
if db_path.is_file() {
   fs::write(&db_path, SECRET_STORE_DB_VERSION).map_err(|e| ...)?;
}
```

Reply:
Shouldn't we check if the folder is empty? Can we trust that old versions of `SecretStore` have created a version file?

Reply:
right, I missed that.

Maybe then do `list.count == 0` instead?

Reply:
I didn't use `count()` because we don't really care how many files we have. 
From the docs:
>This method will call next repeatedly until None is encountered, returning the number of times it saw Some. Note that next has to be called at least once even if the iterator does not have any elements.

Not sure if rust does optimize calls  like `list.count() == 0` but if it doesn't we would iterate over each file in the `iter` where it's enough to find that the first isn't `Some` value.

Reply:
Well, I think you are right because iterators are `lazy`. I think I got it backward and thought count was `memoized`.

Great, thanks for the explanation even though I don't like `iter.next().is_none()` :)

Reply:
I also don't like it. Strangely  there is no `iter.is_emtpy()` or similar.

Reply:
But can you please at least add the comment, explaining, that checks this code? (PR's title as a comment would be sufficient)